crates/nu-command/src/filters/take/take_.rs

```diff
@@ -68,48 +80,92 @@ impl Command for Take {
 
         match input {
             PipelineData::Value(val, _) => match val {
-                Value::List { vals, .. } => Ok(vals
-                    .into_iter()
-                    .take(rows_desired)
-                    .into_pipeline_data(ctrlc)
-                    .set_metadata(metadata)),
+                Value::List { vals, .. } => {
+                    if take_last {
+                        Ok(vals
+                            .into_iter()
+                            .rev()
+                            .take(rows_desired)
+                            .rev()
+                            .into_pipeline_data(ctrlc)
+                            .set_metadata(metadata))
+                    } else {
+                        Ok(vals
+                            .into_iter()
+                            .take(rows_desired)
+                            .into_pipeline_data(ctrlc)
+                            .set_metadata(metadata))
+                    }
+                }
                 Value::Binary { val, span } => {
-                    let slice: Vec<u8> = val.into_iter().take(rows_desired).collect();
+                    let slice: Vec<u8> = if take_last {
+                        val.into_iter().rev().take(rows_desired).rev().collect()
+                    } else {
+                        val.into_iter().take(rows_desired).collect()
+                    };
                     Ok(PipelineData::Value(
                         Value::Binary { val: slice, span },
                         metadata,
                     ))
                 }
-                Value::Range { val, .. } => Ok(val
-                    .into_range_iter(ctrlc.clone())?
-                    .take(rows_desired)
-                    .into_pipeline_data(ctrlc)
-                    .set_metadata(metadata)),
+                Value::Range { val, .. } => {
+                    if take_last {
+                        // TODO: implement DoubleEndedIterator for RangeIterator so we can use .rev().take(n).rev()
+                        // without collecting here
+                        let collected = val.into_range_iter(ctrlc.clone())?.collect_vec();
+                        Ok(collected
+                            .into_iter()
+                            .rev()
+                            .take(rows_desired)
```

Comment:
Here using the ring buffer solution would obviate the need for the superfluous collect as well.